server/sonar-server/src/test/java/org/sonar/server/issue/IssueQueryServiceTest.java

```diff
@@ -434,4 +435,41 @@ public void should_search_on_test_file() throws Exception {
     IssueQuery query = issueQueryService.createFromMap(map);
     assertThat(query.fileUuids()).containsExactly(fileUuid);
   }
+
+  @Test
+  public void fail_if_created_after_and_created_since_are_both_set() {
+    Map<String, Object> map = newHashMap();
+    map.put("createdAfter", "2013-07-25T07:35:00+0100");
+    map.put("createdSince", "palap");
+
+    try {
+      issueQueryService.createFromMap(map);
+      fail();
+    } catch (Exception e) {
+      assertThat(e).isInstanceOf(IllegalArgumentException.class).hasMessage("createdAfter and createdSince cannot be set simultaneously");
+    }
+  }
+
+  @Test
+  public void set_created_after_from_created_since() {
+    Date now = DateUtils.parseDateTime("2013-07-25T07:35:00+0100");
+    when(system.now()).thenReturn(now.getTime());
+    Map<String, Object> map = newHashMap();
+
+    map.put("createdSince", "1d");
+    assertThat(issueQueryService.createFromMap(map).createdAfter()).isEqualTo(DateUtils.parseDateTime("2013-07-24T07:35:00+0100"));
```

Comment:
Don't you think one test is enough, the rest is testing the ISOStandardPeriod formatting